internal/pkg/cli/local_run.go

```diff
@@ -0,0 +1,336 @@
+	}
+
+	ws, err := workspace.Use(afero.NewOsFs())
+	if err != nil {
+		return nil, err
+	}
+	ecsLocalClient := ecs.New(defaultSess)
+	opts := &localRunOpts{
+		localRunVars: vars,
+
+		deployedWkld:       []string{},
+		wkldDeployedToEnvs: make(map[string][]string),
+		prompt:             prompt.New(),
+		store:              store,
+		ws:                 ws,
+		ecsLocalClient:     ecsLocalClient,
+		deployStore:        deployStore,
+	}
+	opts.envVersionGetter = func(envName string) (versionGetter, error) {
+		envDescriber, err := describe.NewEnvDescriber(describe.NewEnvDescriberConfig{
+			App:         opts.appName,
+			Env:         envName,
+			ConfigStore: opts.store,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("new environment compatibility checker: %v", err)
+		}
+		return envDescriber, nil
+	}
+	return opts, nil
+}
+
+func (o *localRunOpts) Validate() error {
+	if o.appName == "" {
+		return errNoAppInWorkspace
+	}
+	if _, err := o.store.GetApplication(o.appName); err != nil {
+		return fmt.Errorf("get application %s: %w", o.appName, err)
+	}
+	return nil
+}
+
+func (o *localRunOpts) Ask() error {
+	if err := o.validateOrAskWorkloadName(); err != nil {
+		return err
+	}
+	if err := o.validateOrAskEnvName(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *localRunOpts) Execute() error {
+	taskDef, err := o.ecsLocalClient.TaskDefinition(o.appName, o.envName, o.wkldName)
+	if err != nil {
+		return fmt.Errorf("get task definition: %w", err)
+	}
+
+	awsSession := session.Must(session.NewSessionWithOptions(
+		session.Options{
+			SharedConfigState: session.SharedConfigEnable,
+		}))
+
+	_, err = o.getDecryptedSSMSecrets(taskDef, awsSession)
+	if err != nil {
+		fmt.Println("unable to get secrets from SSM store")
+	}
+
+	_, err = o.getDecryptedSecretManagerSecrets(awsSession)
+	if err != nil {
+		fmt.Println("unable to get secrets from Secret Manager")
+	}
+
+	return nil
+}
+
+func (o *localRunOpts) getDecryptedSSMSecrets(taskDef *awsecs.TaskDefinition, awsSession *session.Session) ([]ssmSecret, error) {
+	var SSMSecretsList []ssmSecret
+	secrets := taskDef.Secrets()
+	ssmClient := ssm.New(awsSession)
+	for _, secret := range secrets {
+		resp, err :=
+			ssmClient.GetParameter(&ssm.GetParameterInput{
+				Name:           aws.String(secret.ValueFrom),
+				WithDecryption: aws.Bool(true),
+			})
+		if err != nil {
+			return nil, err
+		}
+		secretInfo := ssmSecret{
+			Name:  secret.Name,
+			Value: *resp.Parameter.Value,
+		}
+		SSMSecretsList = append(SSMSecretsList, secretInfo)
+	}
+	return SSMSecretsList, nil
+}
+
+func (o *localRunOpts) getDecryptedSecretManagerSecrets(awsSession *session.Session) ([]secretManager, error) {
+	secretsManagerClient := secretsmanager.New(awsSession)
+	var secretStruct secretManager
+	var SecretManagerSecrets []secretManager
+	input := &secretsmanager.ListSecretsInput{}
+	err := secretsManagerClient.ListSecretsPages(input, func(page *secretsmanager.ListSecretsOutput, lastPage bool) bool {
```

Comment:
Just a question, not a request! Is there a specific reason why we prefer `ListSecretsPages` over `ListSecrets` with pagination token?

Reply:
Correct me if I am wrong. I just thought if there are many secrets using `ListSecretsPages` would just simplify the code which handles pagination internally, reducing the complexity of managing pagination tokens.

Reply:
That SGTM 👍🏼 . Also I just noticed that we are listing all the secrets in the current account + region. Should we just get the value of the secret that found from task definition (like what you did in `getDecryptedSSMSecrets`)